Marlin/src/inc/Conditionals_post.h

```diff
@@ -989,7 +989,14 @@
 #define HAS_AUTO_FAN_5 (HOTENDS > 5 && PIN_EXISTS(E5_AUTO_FAN))
 #define HAS_AUTO_E_FAN (HAS_AUTO_FAN_0 || HAS_AUTO_FAN_1 || HAS_AUTO_FAN_2 || HAS_AUTO_FAN_3 || HAS_AUTO_FAN_4 || HAS_AUTO_FAN_5)
 #define HAS_AUTO_CHAMBER_FAN (HAS_TEMP_CHAMBER && PIN_EXISTS(CHAMBER_AUTO_FAN))
+#define HAS_AUTO_PIN_FAN_0 HAS_AUTO_FAN_0 && (E0_AUTO_FAN_PIN != PS_ON_PIN)
 #define HAS_AUTO_FAN (HAS_AUTO_E_FAN || HAS_AUTO_CHAMBER_FAN)
+#define HAS_AUTO_PIN_FAN_1 HAS_AUTO_FAN_1 && (E1_AUTO_FAN_PIN != PS_ON_PIN)
+#define HAS_AUTO_PIN_FAN_2 HAS_AUTO_FAN_2 && (E2_AUTO_FAN_PIN != PS_ON_PIN)
+#define HAS_AUTO_PIN_FAN_3 HAS_AUTO_FAN_3 && (E3_AUTO_FAN_PIN != PS_ON_PIN)
+#define HAS_AUTO_PIN_FAN_4 HAS_AUTO_FAN_4 && (E4_AUTO_FAN_PIN != PS_ON_PIN)
+#define HAS_AUTO_PIN_FAN_5 HAS_AUTO_FAN_5 && (E5_AUTO_FAN_PIN != PS_ON_PIN)
+#define HAS_AUTO_CHAMBER_PIN_FAN_0 HAS_AUTO_CHAMBER_FAN && (CHAMBER_AUTO_FAN_PIN != PS_ON_PIN)
```

Comment:
It looks to me like you could just modify the definition of `HAS_AUTO_FAN_n`, etc., adding on `(En_AUTO_FAN_PIN != PS_ON_PIN)` and so on, and the behavior would be correct everywhere.

Reply:
Thanks, thats a good idea, I will test it out again later.

Reply:
Thanks, I did your suggestion and it works. Removed the changes to temperature.cpp for the extruder fans and chamber fans.
I did a similar change for the controller fan, I updated the macros in conditional_post.h with similar checks to the extruder fans. I did have to change to use the HAS_CONTROLER_FAN macro rather than USE_CONTROLER_FAN in a couple of files.
I compiled the code, with and without auto power and with and without autofan and with/without autofan pin = PS_ON_PIN.
Testing - I am using a delta - dont think it makes any difference.
1. Without this change, if I heat the extruder up ~80c, settarget to 0, wait 30seconds, extruder fan turned off and extruder temp was > 50C
2. With this change, heat extruder up ~80C, settarget to 0, wait 30sec, extruder fan is still running, 30sec after extruder temp < 50C power turns off and extruder fan stops.
3. With controllerfan, run axis home, power turns on, controllerfan on, 30 sec after homing finishes, power still on. disable steppers (M18), 30secs power turns off and controllerfan turns off.